yarf/rf_libraries/libraries/image/segmentation.py

```diff
@@ -0,0 +1,255 @@
+import cv2
+import numpy as np
+from PIL import Image
+from robot.api import logger
+
+from yarf.vendor.RPA.Images import RGB
+
+
+class SegmentationTool:
+    """
+    Class implementing segmentation tools used to match text with colors.
+    """
+
+    def is_hsv_color_similar(
+        self, color1: tuple[int], color2: tuple[int], tolerance: int
+    ):
+        """
+        Check if two colors are similar in HSV color space.
+
+        Compares two colors and determines if they are within the specified tolerance.
+
+        Args:
+            color1: First color in HSV format (h, s, v)
+            color2: Second color in HSV format (h, s, v)
+            tolerance: Allowed difference in % (default: 20)
+
+        Returns:
+            True if colors are similar within tolerance, False otherwise
+        """
+        for i in (0, 2):  # ignoe saturation
+            max = int(color1[i] + (255.0 * tolerance / 100.0))
+            min = int(color1[i] - (255.0 * tolerance / 100.0))
+            logger.info(
+                "Checking if {}={} is between {} and {}".format(
+                    ("H" if i == 0 else "V"), color2[i], min, max
+                )
+            )
+            if max > color2[i] and min < color2[i]:
```

Comment:
```suggestion
            if min < color2[i] < max:
```

Your call on this one.